iceoryx_posh/include/iceoryx_posh/popo/request.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+// Copyright (c) 2021 by AVIN Systems Private Limited All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_REQUEST_HPP
+#define IOX_POSH_POPO_REQUEST_HPP
+
+#include "iceoryx_posh/popo/smart_chunk.hpp"
+
+namespace iox
+{
+namespace popo
+{
+template <typename T>
+class RpcInterface;
+
+///
+/// @brief The Request class is a mutable abstraction over types which are written to loaned shared memory.
+/// These requests are transmittable to the iceoryx system.
+///
+template <typename T>
+class Request : public SmartChunk<RpcInterface<Request<T>>, T, RequestHeader>
```

Comment:
I think once you create a test for this class, this won't compile since there is no `RpcInterface` defined anywhere.

Also, at some places you use `RcpInterface<T>` which is passed to `Base_t` which has a `RpcInterface<Request<T>>`.

Could you please create some tests like in `test_popo_sample.cpp` but for `Request` and `Response`